File: Java/프로그래머스/1/12915. 문자열 내 마음대로 정렬하기/문자열 내 마음대로 정렬하기.java
import java.util.*;

class Solution {
    public String[] solution(String[] strings, int n) {
        Arrays.sort(strings);
        String temp = "";
        for(int i = 0; i < strings.length-1; i++) {
            for(int j= 1 ; j < strings.length-i; j++) {
                if(strings[j-1].charAt(n) > strings[j].charAt(n)) {
                    temp = strings[j-1];
                    strings[j-1] = strings[j];
                    strings[j] = temp;
                }
            }
        }
        
        return strings;
    }
}

File: Kotlin/프로그래머스/1/12915. 문자열 내 마음대로 정렬하기/문자열 내 마음대로 정렬하기.kt
class Solution {
    fun solution(strings: Array<String>, n: Int): Array<String> {
        var answer = Array(strings.size) { i -> strings[i][n].toString() }
        answer.sort()
        var t = strings.toMutableList()
            t.sort()
        for(j in answer.indices){
            var i = 0
            while(t.size>i) {
                if(answer[j]==t[i][n].toString()) {
                    answer[j] = t[i]
                    t.removeAt(i)
                } else {
                    i++
                }
            }
        }
        
        return answer
    }
}